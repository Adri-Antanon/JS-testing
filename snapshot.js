export const getCharacters = (data) => ({
  id: data.id,
  name: data.name,
  status: data.species,
  gender: data.gender,
});
