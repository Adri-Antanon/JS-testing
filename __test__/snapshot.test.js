import { getCharacters } from "../snapshot";
import rick from "../rick.json";

describe("It's time for the snapshots", () => {
  test("Snapshot ", () => {
    expect(getCharacters(rick)).toMatchSnapshot();
  });

  test("should fail the snapshot", () => {
    const user = {
      createAt: new Date(),
      id: Math.floor(Math.random() * 20),
    };

    expect(user).toMatchSnapshot();
  });

  test("should have an exception", () => {
    const user = {
      name: "Adri",
      id: Math.floor(Math.random() * 20),
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    });
  });
});
