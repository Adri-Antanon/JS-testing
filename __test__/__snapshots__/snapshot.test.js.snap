// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It's time for the snapshots Snapshot  1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "status": "Human",
}
`;

exports[`It's time for the snapshots should fail the snapshot 1`] = `
Object {
  "createAt": 2021-06-14T07:49:31.122Z,
  "id": 17,
}
`;

exports[`It's time for the snapshots should have an exception 1`] = `
Object {
  "id": Any<Number>,
  "name": "Adri",
}
`;
